lib/FlashMem/FlashMem.cpp

```diff
@@ -0,0 +1,191 @@
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+/*******************************************************************************
+    DESCRIPTION
+*******************************************************************************/
+/**
+ * @brief  Implememtation of FlashMem.h
+ * @author Gabryel Reyes <gabryelrdiaz@gmail.com>
+ */
+
+/******************************************************************************
+ * Includes
+ *****************************************************************************/
+#include "FlashMem.h"
+
+/******************************************************************************
+ * Compiler Switches
+ *****************************************************************************/
+
+/******************************************************************************
+ * Macros
+ *****************************************************************************/
+
+/******************************************************************************
+ * Types and classes
+ *****************************************************************************/
+
+/******************************************************************************
+ * Prototypes
+ *****************************************************************************/
+
+/******************************************************************************
+ * Local Variables
+ *****************************************************************************/
+
+/******************************************************************************
+ * Public Methods
+ *****************************************************************************/
+
+bool Flash::begin()
+{
+    bool isSuccess = true;
+    EEPROM.begin(EEPROM_SIZE);
+    return isSuccess;
+}
+
+bool Flash::importCredentials(String &ssid, String &password)
+{
+    bool isSuccess = false;
+
+    if (areCredentialsStored())
+    {
+        char temp = 0x00;
+
+        for (uint8_t memoryPosition = 0; memoryPosition <= NVM_SSID_MAX_LENGTH; memoryPosition++)
+        {
+            temp = EEPROM.read(NVM_SSID_ADDRESS + memoryPosition);
+
+            if (0x00 == temp)
+            {
+                break;
+            }
+            else
+            {
+                ssid += temp;
+            }
+        }
+
+        if (0x00 != EEPROM.read(NVM_PASSWORD_ADDRESS))
```

Comment:
Why this is necessary here, but not for SSID?

Reply:
For SSID, the function `areCredentialsStored()` do this check, as there must be an SSID to connect to. 
However there may not always be a Password if the Network is unsecured, that is why here it does not fail when no password is found. 
Or are we closing the possibility of connecting to unsecured networks (thinking about Eduroam for example)?

Reply:
I think this check is not necessary, because the following for loop see the string termination as well and breaks.